### PR TITLE
tls-sni-01

### PR DESCRIPTION
**Pull request by kuba**
Another step towards #914.

### CODE REVIEW

File: acme/acme/challenges.py [hunk 1]
```diff
@@ -318,89 +319,36 @@ def validation(self, account_key):
...
+    typ = "tls-sni-01"
 
     DOMAIN_SUFFIX = b".acme.invalid"
     """Domain name suffix."""
 
-    PORT = DVSNI.PORT
-    """Port to perform DVSNI challenge."""
-
-    validation = jose.Field("validation", decoder=jose.JWS.from_json)
+    PORT = 443
```
Comment (pde) [inline acme/acme/challenges.py]: Is hardcoding this going to mess with inegration tests?

File: acme/acme/challenges.py [hunk 2]
```diff
@@ -318,89 +319,36 @@ def validation(self, account_key):
...
-
-        """
-        # Instance of 'Field' has no 'signature' member
-        # pylint: disable=no-member
-        return hashlib.sha256(self.validation.signature.encode(
-            "signature").encode("utf-8")).hexdigest().encode()
+    def z(self):
+        """``z`` value used for verification."""
+        return hashlib.sha256(
+            self.key_authorization.encode("utf-8")).hexdigest().encode()
```
Comment (pde) [inline acme/acme/challenges.py]: Note: we assume `hexidgest` is always lowercase, which is probably fine but may deserve a comment.

**Comment by pde** [comment]
Otherwise, LGTM.

File: acme/acme/challenges.py (hunk 1 above)
Comment (kuba) [inline acme/acme/challenges.py]: No, this the default ACME port. You can pass `port` to `simple_verify` to override it. Added an inline comment.

File: acme/acme/challenges.py (hunk 2 above)
Comment (kuba) [inline acme/acme/challenges.py]: Interestingly, Python docs doesn't guarantee it to be lower case, so I've added `.lower()`.

https://docs.python.org/2/library/hashlib.html#hashlib.hash.hexdigest

**Comment by kuba** [comment]
Fixed comments. Ready to merge.

**Pull request merged**